### PR TITLE
Protected pages

### PR DESCRIPTION
**Pull request by Monnorcurphy**
Thomas, before we merge this be aware of the following:

You will have to login in order to edit feed and any other protected pages
You will need the env file.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,3903 +1,6 @@
...
-      "integrity": "sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==",
-      "engines": {
-        "node": ">=10"
-      },
-      "funding": {
-        "url": "https://github.com/sponsors/sindresorhus"
-      }
-    }
-  },
+  "lockfileVersion": 1,
```
Comment (indifferentghost) [inline package-lock.json:3]: If you don't update your npm this discrepency will keep happening.

File: pages/index.js [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+  if (isLoading) return <div>Loading...</div>;
+  if (error) return <div>{error.message}</div>;
+
+  if (user) {
+    router.push("/feed");
+  }
+
+ return (
+   <div>
+     <a href="/api/auth/login">Login</a>
```
Comment (indifferentghost) [inline pages/index.js:24]: Upon trying to click this I get `"secret" is required`

File: components/protected.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import { useRouter } from "next/router";
+
+//check if you are on the client (browser) or server
```
Comment (indifferentghost) [inline components/protected.js]: Doesn't the name indicate this?
```suggestion
```

File: components/protected.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const { isAuthenticated } = useAuth0();
+  const router = useRouter();
+
+  let unprotectedRoutes = [
+      '/about',
+      '/legal',
+      '/'
+  ];
+
+  debugger;
```
Comment (indifferentghost) [inline components/protected.js]: Oooooo debuggers in PRs;
```suggestion
```

File: pages/_app.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+import { UserProvider } from "@auth0/nextjs-auth0";
+import { Auth0Provider } from '@auth0/auth0-react';
+
 
 function MyApp({ Component, pageProps }) {
-  return(
-  <Auth0Provider
-    domain="dev-0tw3ahw4.us.auth0.com"
-    clientId="znaCBTn6YtR8KYMMYC0rHVwXmR9b0iWF"
-    redirectUri='http://localhost:3000'
```
Comment (indifferentghost) [inline pages/_app.js:9]: How are these getting set now?

File: pages/index.js [hunk 6]
```diff
@@ -1,12 +1,39 @@
...
-    <div>TODO (maybe redirect to /feed if logged in)
-      <LoginButton/>
+
+
+export default function Index() {
+  const { user, error, isLoading } = useUser();
+  const router = useRouter();
+
+  if (isLoading) return <div>Loading...</div>;
+  if (error) return <div>{error.message}</div>;
```
Comment (indifferentghost) [inline pages/index.js:16]: Wait if it errors shouldn't we just display the home page?

File: pages/profile.js [hunk 7]
```diff
@@ -1,24 +1,27 @@
...
+        <div className="max-w-3xl mx-auto divide-y-2 divide-gray-200">
+          <dl className="mt-6 space-y-6 divide-y divide-gray-200">
+            <img src={user.picture} alt={user.name} />
+            <h2>{user.name}</h2>
+            <p>{user.email}</p>
+          </dl>
+        </div>
+      </Layout>
+    );
   }
```
Comment (indifferentghost) [inline pages/profile.js]: There doesn't need to be an else here.
```suggestion
  }
  return (
    <Layout title="Profile">
      <div className="max-w-3xl mx-auto divide-y-2 divide-gray-200">
        <dl className="mt-6 space-y-6 divide-y divide-gray-200">
          <img src={user.picture} alt={user.name} />
          <h2>{user.name}</h2>
          <p>{user.email}</p>
        </dl>
      </div>
    </Layout>
  );
```

File: components/protected.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import { useRouter } from "next/router";
+
+//check if you are on the client (browser) or server
+const isBrowser = () => typeof window !== "undefined";
+
+const ProtectedRoute = ({ children }) => {
```
Comment (indifferentghost) [inline components/protected.js]: This doesn't get used anywhere. We are going to need something like this for displaying some features. (should probably have a nav on public pages so users can easily get to their feed.

Comment (indifferentghost) [reply components/protected.js]: Maybe? OF doesn't do this, I wonder if it's a legal reason. (Infact they display a basic html page.

File: components/layout.js [hunk 9]
```diff
@@ -6,20 +6,34 @@ import {
...
-  { name: "About", href: "/about", icon: QuestionMarkCircleIcon, current: false },
+  {
+    name: "About",
+    href: "/about",
+    icon: QuestionMarkCircleIcon,
+    current: false,
+  },
   { name: "Profile", href: "/profile", icon: UserCircleIcon, current: false },
   { name: "Messages", href: "/messages", icon: ChatAlt2Icon, current: false },
+  { name: "Legal", href: "/legal", icon: BriefcaseIcon, current: false },
```
Comment (indifferentghost) [inline components/layout.js:27]: I don't think this should belong on the nav.

File: components/layout.js [hunk 10]
```diff
@@ -6,20 +6,34 @@ import {
...
+  {
+    name: "Log out",
+    href: "/api/auth/logout",
+    icon: LogoutIcon,
+    current: false,
+  },
```
Comment (indifferentghost) [inline components/layout.js:33]: I don't think this should be on the nav

File: pages/index.js [hunk 11]
```diff
@@ -1,12 +1,39 @@
...
+
+  if (user) {
+    router.push("/feed");
+  }
```
Comment (indifferentghost) [inline pages/index.js:20]: It might be worthwhile to have the homepage/feed both be on root -> thinking of how EVERYONE ELSE does it.

**Review by indifferentghost — CHANGES_REQUESTED** [review]
The only major change I'm focused on is that the login is broken (for me) and I can't get to my feed w/o commenting code.

File: pages/_app.js (hunk 5 above)
Comment (Monnorcurphy) [reply pages/_app.js:9]: In the secret file <3

File: pages/index.js (hunk 11 above)
Comment (Monnorcurphy) [reply pages/index.js:20]: Yeah, I agree with that.

File: components/layout.js (hunk 10 above)
Comment (Monnorcurphy) [reply components/layout.js:33]: Where should it be?

File: components/layout.js (hunk 9 above)
Comment (Monnorcurphy) [reply components/layout.js:27]: On the bottom then? (i.e. a footer?)

Comment (indifferentghost) [reply components/layout.js:27]: Yes a footer would be better.

**Review by indifferentghost — APPROVED** [review]

**Pull request merged**